src/main/java/com/sphy/hotelmanagementapplication/service/HotelService.java

```diff
@@ -237,4 +239,22 @@ public List<HotelDTO> saveHotels(List<HotelDTO> hotelsDTO) throws ApiRequestExce
 		return hotelDTOS;
 	}
 
+
+	/***
+	 * Retrieves a set of HotelAmenity for the hotel with the given ID
+	 * @param id The ID of the hotel for which to retrieve amenities
+	 * @return A set of HotelAmenity representing the amenities of the hotel with the given ID
+	 */
+
+	public Set<HotelAmenityDTO> getHotelAmenitiesByHotelId(Long id){
+
+		Set<HotelAmenityDTO> amenitiesHotelDTO = new HashSet<>();
```

Comment:
replace all method with this : Set<HotelAmenityDTO> amenitiesHotelDTO = new HashSet<>();

		Optional<Hotel> hotelOptional = hotelRepository.findById(id);

		hotelOptional.ifPresent(hotel -> hotel.getHotelAmenity()
				.forEach(hotelAmenity -> {
					amenitiesHotelDTO.add(hotelAmenityToHotelAmenityDTO.converter(hotelAmenity));
				}));
		return amenitiesHotelDTO;